### PR TITLE
Fix euler1d not responding on Windows

### PR DESCRIPTION
**Pull request by tychuang1211**
This PR is a simple fix for #334.

According to #334, under some conditions `sys.stdout` can be `None`, which is usually the case for Windows GUI apps that aren't connected to a console.

Added a single line to check the value of `sys.stdout` before writing to fix the issue.

### CODE REVIEW

File: modmesh/app/euler1d.py [hunk 1]
```diff
@@ -727,8 +727,9 @@ def log(msg):
 
         :return: None
         """
-        sys.stdout.write(msg)
-        sys.stdout.write('\n')
+        if sys.stdout != None:
```
Comment (yungyuc) [inline modmesh/app/euler1d.py]: What about adding a comment pointing to https://github.com/solvcon/modmesh/issues/334 ?  We may need some refactoring before being able to make a regression (unit) test.  Some explanation in code will help reduce the chance of regression before that.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Thanks @tychuang1211 .  A minor request:

- [x] Add comments.

File: modmesh/app/euler1d.py (hunk 1 above)
Comment (tychuang1211) [reply modmesh/app/euler1d.py]: Added!

**Comment by tychuang1211** [comment]
@yungyuc I added the comment.

**Review by yungyuc — APPROVED** [review]
Thanks, @tychuang1211 !

**Pull request merged**